### PR TITLE
Improve block positioning logic

### PR DESCRIPTION
**Pull request by Hamms**
The block positioning logic had an issue with aboslutely-positioned
non-user-visible blocks not ending up in the right place. It had this
bug in part because the logic of this section has become extremely
messy, and in part because the tests were insufficient.

This addresses both of those problems.

### CODE REVIEW

File: apps/test/solutions/maze/hidden_start_blocks.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
..."DIR">moveForward</title><next><block type="controls_repeat" uservisible="false"><title name="TIMES">3</title><statement name="DO"><block type="maze_nectar" uservisible="false"/></statement><next><block type="maze_move" uservisible="false"><title name="DIR">moveBackward</title><next><block type="maze_move" uservisible="false"><title name="DIR">moveBackward</title><next><block type="controls_if" inline="false" uservisible="false"><mutation else="1"/><value name="IF0"><block type="logic_compare" inline="true" movable="false" uservisible="false"><title name="OP">EQ</title><value name="A"><block type="variables_get" movable="false" uservisible="false"><title name="VAR">direction</title></block></value><value name="B"><block type="variables_get" uservisible="false"><title name="VAR">left</title></block></value></block></value><statement name="DO0"><block type="maze_turn" uservisible="false"><title name="DIR">turnRight</title></block></statement><statement name="ELSE"><block type="maze_turn" uservisible="false"><title name="DIR">turnLeft</title></block></statement></block></next></block></next></block></next></block></next></block></next></block></next></block></statement></block></xml>'
```
Comment (Bjvanminnen) [inline apps/test/solutions/maze/hidden_start_blocks.js:46]: Might be worth a comment about what's interesting about this xml (I think it's that y is -5 for our first when_run)

**Comment by Bjvanminnen** [comment]
LGTM. nice work tracking down this tricksy bug

File: apps/test/solutions/maze/hidden_start_blocks.js (hunk 1 above)
Comment (Hamms) [inline apps/test/solutions/maze/hidden_start_blocks.js:46]: I've got a comment above the levelDef; is that not sufficiently clear?

**Comment by Hamms** [comment]
Thanks!

File: apps/test/solutions/maze/hidden_start_blocks.js (hunk 1 above)
Comment (Bjvanminnen) [inline apps/test/solutions/maze/hidden_start_blocks.js:46]: My bad. That comment seems sufficient.

**Pull request merged**